elastic/kibana
# [Guided onboarding] Update observability tour

This PR updates the [Observability product tour implemented in 8.4](https://github.com/elastic/kibana/pull/133909) to work with the new guided onboarding framework. The tour will now display as part of step 3 of the Kubernetes guide.

Fixes https://github.com/elastic/kibana/issues/140656

### How to test
1. Enable guided onboarding via your `kibana.dev.yml` file: `guidedOnboarding.ui: true`
2. Start up Kibana with the example plugins enabled: `yarn start --run-examples`
3. Navigate to `/app/guidedOnboardingExample`
4. Use the "Set guide state" form to trigger step 3 of the Observability guide. Click save after setting the following values:
    ```
    Guide: observability
    Step ID: tour_observability
    ```
5. Start the step. Verify the guide navigates the user to the Observability Overview page and the EuiTour displays.
7. Complete the tour. Verify the step is completed and the guide panel reopens.
8. All other tour behavior implemented via https://github.com/elastic/kibana/pull/133909 should largely remain. Skipping a tour should complete the guide. If you leave the tour mid-tour and come back, the correct step should display. Clicking on the "Data assistant" button should also complete the guide.

**Note:** There is a bug right now when clicking on other links in the side nav with the tour in progress. This appears to be a regression introduced in the `KibanaPageTemplate`. I opened https://github.com/elastic/kibana/issues/143114 to track.

### Demo
https://user-images.githubusercontent.com/5226211/195176561-f92a76ea-1a43-4230-9665-ef971bced502.mp4
